### PR TITLE
[Issue 9297][broker] Try to fix flaky infinite reconnection

### LINKED ISSUE #9297: proxy lookup result still contains unreachable owner broker serviceURL after broker restart and 30s zk session expired
**Describe the bug**
- Certain issue
  After broker restarted with 30s zk session expired, client take 51s, even 1min40s to reconnect successfully, recovery time is a bit long. 

- Flaky issue
  Client reconnect always failed handshake with proxy, stucked at 1~5

image

**To Reproduce**
1. use pulsar-helm-chart to deploy [pulsar-all:2.6.0](https://hub.docker.com/layers/apachepulsar/pulsar-all/2.6.0/images/sha256-d93e9ac275fe0a6f71d0fd4c9e68d2efb64d4c6fa165c1c07e6c0928d7762fa5?context=explore) cluster
- 1\*zk, 3\*brokers, 3\*bookies, 3\*proxy, all with 2c 2g resource limit
- 4k QPS, 4MBps In/out, for cpu and memory, load < 35%, see [3_brokers_loadreport.json.log](https://github.com/apache/pulsar/files/5862180/3_brokers_loadreport.json.log)

2. Manually delete one broker and restart, it's owning topics(bundles) will be unloaded, clients disconnect with proxy and reconnect, 0.1s, 0.2s, 0.4s, ..., 25.6s at least needed, 51.2s is common.
In flaky case, client reconnect always failed, it has been seen in production env, but it's not easy to reproduce.

**Expected behavior**
After broker restart, client need reconnect succeed as soon as possible, like within 40s(30s zk session expiration + 10s reconnect)

**Screenshots**
image

**Other**
How to fix certain case to reduce recovery time?
- Reduce zk session expiration time, default 30s, reduce to 10s, but with what cost?
- Client backoff policy default exponential, may need provide a option to limit the backoff limit, such as 5s, backoff should be 0.1s, 0.2s, 0.4s, ...,3.2s, 5s, 5s, ...
- Reduce broker znode expiration waiting time, default 10s, reduce to 2s, see [broker-statefulset.yaml#L197](https://github.com/apache/pulsar-helm-chart/blob/master/charts/pulsar/templates/broker-statefulset.yaml#L197)

How to fix flaky case?
I tried to fix in [PR#9298](https://github.com/apache/pulsar/pull/9298), but I'm not sure it covered this case.

### PR DESCRIPTION
**Pull request by wuYin**
Fixes #9297 

### Motivation

Try to fix flaky case which broker always returned unreachable address to client.

### Modifications

Before broker response lookup result, check the owner broker is active or not.
If not, remove the ownership, then redirect to leader broker to find a new active broker to own this bundle. 

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

### Other
Since this case can't be reproduced easily, and this change will cause the lookup to increase the latency of the read local cache, so I'm not sure this is the best solution to fix this, if not, ok to close this PR.

### CODE REVIEW

**Review by jiazhai — APPROVED** [review]

**Comment by zymap** [comment]
Looks like tests failed by this change:
```
Error:  Failures: 
Error:  org.apache.pulsar.broker.namespace.OwnershipCacheTest.testGetOwner(org.apache.pulsar.broker.namespace.OwnershipCacheTest)
[INFO]   Run 1: PASS
Error:    Run 2: OwnershipCacheTest.testGetOwner:202 ? Execution java.lang.NullPointerException
[INFO] 
Error:  org.apache.pulsar.broker.namespace.OwnershipCacheTest.testReestablishOwnership(org.apache.pulsar.broker.namespace.OwnershipCacheTest)
[INFO]   Run 1: PASS
Error:    Run 2: OwnershipCacheTest.testReestablishOwnership:354 ? Execution java.lang.NullPoin...
[INFO] 
Error:    TransactionMetaStoreAssignmentTest.init:34->TransactionMetaStoreTestBase.setup:53 ? SessionExpired
[INFO] 
Error:  Tests run: 499, Failures: 3, Errors: 0, Skipped: 1
```

**Comment by wuYin** [comment]
@zymap thanks for review
Actually, I'm not sure this ownership double check PR will solve the flaky Issue
I'll add test to simulate flaky case by removing the /loadbalance/downBroker zNode directly, and fix existed failure tests

**Comment by jiazhai** [comment]
@wuYin this change lgtm. the latency for read local zk cache should be fine. Otherwise it would be hard to fix this issue. 
It would be great to add a test to cover and protect this change.

**Review by sijie — APPROVED** [review]

**Review by codelipenghui — APPROVED** [review]

**Comment by codelipenghui** [comment]
@wuYin Could you please help check the failed tests? I have moved it to 2.9.0 and tag `release/2.8.1`, if the test passed, we can move it back to 2.8.0

**Comment by hangc0276** [comment]
@wuYin  Would you please address the conflict? i am starting to release 2.8.1, it all tests passed, we can include this pr in 2.8.1

**Comment by hangc0276** [comment]
@wuYin Could you please help check the failed tests and the conflicts?  I have moved it to  tag release/2.8.2,  if the test passed and conflict addressed, we can move it back to 2.8.1.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/OwnershipCache.java [hunk 1]
```diff
@@ -238,7 +239,32 @@ public OwnershipCache(PulsarService pulsar, NamespaceBundleFactory bundleFactory
...
+            } else {
+                try {
+                    URI ownerURI = new URI(owner.get().getHttpUrl());
+                    String ownerServiceAddr = ownerURI.getHost() + ":" + ownerURI.getPort();
+                    if (pulsar.getLoadManager().get().getAvailableBrokers().contains(ownerServiceAddr)) {
+                        ownerFuture.complete(owner);
+                    } else {
+                        LOG.warn("Namespace {} owned by broker {} which is not active, remove the ownership",
+                                suName.toString(), owner);
+                        removeOwnership(suName);
```
Comment (eolivelli) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/OwnershipCache.java:257]: here we are in `getOwnerAsync` that is meant to be free from side effects.

I am not sure this is the right fix

**Review by eolivelli — CHANGES_REQUESTED** [review]

**Comment by 315157973** [comment]
@wuYin  Hello, please take a look at eolivelli's comments

**Pull request closed without merging**